File: andtabs/src/main/java/com/devmichaelgirgis/andtabs/customview/EasyTabs.java
package com.devmichaelgirgis.andtabs.customview;


import android.content.Context;
import com.google.android.material.tabs.TabLayout;
import androidx.viewpager.widget.ViewPager;
import android.util.AttributeSet;
import android.view.View;
import android.widget.RelativeLayout;

import com.devmichaelgirgis.andtabs.R;


/**
 * Created by gilgoldzweig on 24/10/2016.
 * Enjoy
 */

public class EasyTabs extends RelativeLayout {
    private TabLayout tabs;
    private ViewPager pager;
    private View rootView;

    public EasyTabs(Context context) {
        super(context);
        rootView = inflate(context, R.layout.easytabs, this);
        tabs = rootView.findViewById(R.id.tabs);
        pager = rootView.findViewById(R.id.pager);


    }

    public EasyTabs(Context context, AttributeSet attrs) {
        super(context, attrs);
        rootView = inflate(context, R.layout.easytabs, this);
        tabs = rootView.findViewById(R.id.tabs);
        pager = rootView.findViewById(R.id.pager);

    }

    public EasyTabs(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        rootView = inflate(context, R.layout.easytabs, this);
        tabs = rootView.findViewById(R.id.tabs);
        pager = rootView.findViewById(R.id.pager);
    }


    public ViewPager getViewPager() {
        return pager;
    }

    public TabLayout getTabLayout() {
        return tabs;
    }
}

File: andtabs/src/main/java/com/devmichaelgirgis/andtabs/customview/EasyTabs.kt
package com.devmichaelgirgis.andtabs.customview


import android.content.Context
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import android.util.AttributeSet
import android.view.View
import android.widget.RelativeLayout

import com.devmichaelgirgis.andtabs.R



class EasyTabs : RelativeLayout {
    var tabLayout: TabLayout? = null
        private set
    var viewPager: ViewPager? = null
        private set
    private var mRootView: View? = null

    constructor(context: Context) : super(context) {
        mRootView = View.inflate(context, R.layout.easytabs, this)
        tabLayout = mRootView!!.findViewById(R.id.tabs)
        viewPager = mRootView!!.findViewById(R.id.pager)


    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        mRootView = View.inflate(context, R.layout.easytabs, this)
        tabLayout = mRootView!!.findViewById(R.id.tabs)
        viewPager = mRootView!!.findViewById(R.id.pager)

    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        mRootView = View.inflate(context, R.layout.easytabs, this)
        tabLayout = mRootView!!.findViewById(R.id.tabs)
        viewPager = mRootView!!.findViewById(R.id.pager)
    }
}
